hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1060,11 +1060,19 @@ def _from_type_deferred(thing: Type[Ex]) -> SearchStrategy[Ex]:
     # underlying strategy wherever possible, as a form of user education, but
     # would prefer to fall back to the default "from_type(...)" repr instead of
     # "deferred(...)" for recursive types or invalid arguments.
+    if (
+        thing.__repr__ == object.__repr__
+        and hasattr(thing, "__module__")
+        and hasattr(thing, "__qualname__")
+    ):
+        thing_repr = f"{thing.__module__}.{thing.__qualname__}"
+    else:
+        thing_repr = repr(thing)
     return LazyStrategy(
         lambda thing: deferred(lambda: _from_type(thing, [])),
         (thing,),
         {},
-        force_repr=f"from_type({thing!r})",
+        force_repr=f"from_type({thing_repr})",
```

Comment:
I think we could use `nicerepr()` from:
https://github.com/HypothesisWorks/hypothesis/blob/de05c94c002becadd5ca6ec847fe86cae7e27688/hypothesis-python/src/hypothesis/internal/reflection.py#L445-L452

```suggestion
        force_repr=f"from_type({nicerepr(thing)})",
```
which will display it the same way as in various other contexts, including handling typevars.

Reply:
I was not aware of `nicerepr`, that's very nice! Even though I had to add a workaround to add the module name if it was not already added.